GVRf/Extensions/gvrf-physics/src/main/java/org/gearvrf/physics/GVRWorld.java

```diff
@@ -113,9 +117,15 @@ public void removeBody(GVRRigidBody gvrBody) {
 
     @Override
     public void onDrawFrame(float frameTime) {
-        NativePhysics3DWorld.step(getNative(), frameTime);
-
+        mFrameTime += frameTime;
+        if (mIsProcessing) {
+            return;
+        }
+        mIsProcessing = true;
+        NativePhysics3DWorld.step(getNative(), mFrameTime);
```

Comment:
The draw frame listener is called pretty frequently. If you launch a lot of balls at the same time or the collisions are complex, the physics will not finish before the next call is issued. Also if you have more than one physics world there will be multiple draw frame listeners for step() and calls could happen at the same time.
This fix is not ideal and we need to revisit the physics thread safety issues.